hack/local-up-cluster.sh

```diff
@@ -399,47 +399,52 @@ cleanup()
 # a process dies unexpectedly.
 function healthcheck {
   if [[ -n "${APISERVER_PID-}" ]] && ! sudo kill -0 ${APISERVER_PID} 2>/dev/null; then
-    warning "API server terminated unexpectedly, see ${APISERVER_LOG}"
+    warning_log "API server terminated unexpectedly, see ${APISERVER_LOG}"
     APISERVER_PID=
   fi
 
   if [[ -n "${CTLRMGR_PID-}" ]] && ! sudo kill -0 ${CTLRMGR_PID} 2>/dev/null; then
-    warning "kube-controller-manager terminated unexpectedly, see ${CTLRMGR_LOG}"
+    warning_log "kube-controller-manager terminated unexpectedly, see ${CTLRMGR_LOG}"
     CTLRMGR_PID=
   fi
 
   if [[ -n "$DOCKERIZE_KUBELET" ]]; then
     # TODO (https://github.com/kubernetes/kubernetes/issues/62474): check health also in this case
     :
   elif [[ -n "${KUBELET_PID-}" ]] && ! sudo kill -0 ${KUBELET_PID} 2>/dev/null; then
-    warning "kubelet terminated unexpectedly, see ${KUBELET_LOG}"
+    warning_log "kubelet terminated unexpectedly, see ${KUBELET_LOG}"
     KUBELET_PID=
   fi
 
   if [[ -n "${PROXY_PID-}" ]] && ! sudo kill -0 ${PROXY_PID} 2>/dev/null; then
-    warning "kube-proxy terminated unexpectedly, see ${PROXY_LOG}"
+    warning_log "kube-proxy terminated unexpectedly, see ${PROXY_LOG}"
     PROXY_PID=
   fi
 
   if [[ -n "${SCHEDULER_PID-}" ]] && ! sudo kill -0 ${SCHEDULER_PID} 2>/dev/null; then
-    warning "scheduler terminated unexpectedly, see ${SCHEDULER_LOG}"
+    warning_log "scheduler terminated unexpectedly, see ${SCHEDULER_LOG}"
     SCHEDULER_PID=
   fi
 
   if [[ -n "${ETCD_PID-}" ]] && ! sudo kill -0 ${ETCD_PID} 2>/dev/null; then
-    warning "etcd terminated unexpectedly"
+    warning_log "etcd terminated unexpectedly"
     ETCD_PID=
   fi
 }
 
 function warning {
   message=$1
+  prefix=$2
 
   echo $(tput bold)$(tput setaf 1)
-  echo "WARNING: ${message}"
+  echo "${prefix}WARNING: ${message}"
   echo $(tput sgr0)
 }
 
+function warning_log {
+  warning "$1" "[$(date)] "
```

Comment:
What else writes timestamps this way? Instead can we follow the same pattern used elsewhere?

For example in [our e2e testing](https://github.com/kubernetes/test-infra/blob/738ce9c95dd9afa2d94bb77d571e68a8bc388cc8/jenkins/bootstrap.py#L778):

```
 # [IWEF]mmdd HH:MM:SS.mmm] msg
 # aka
I0628 17:39:50.369] STEP: Creating a kubernetes client
```

Another alternative is what we see [from ginkgo logs](https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/65374/pull-kubernetes-e2e-gce/42450/?log#log):
```
# Mon dd HH:MM:SS.uuu: INFO
# Aka
Jun 28 17:37:43.554: INFO: PersistentVolumeClaim pvc-2snmq found but phase is Pending instead of Bound.
```

Reply:
The date format here looks good to me: 
https://github.com/kubernetes/test-infra/blob/738ce9c95dd9afa2d94bb77d571e68a8bc388cc8/jenkins/bootstrap.py#L780
While further thinking about it, I added a generic printer to support other colors as well!